File: 1/solution.java
class Solution {
    public boolean isPalindrome(int number) {
        if(number<0) return false;
        else if(number<10) return true;
        else if(number%10==0) return false;
        else if(number<100&&number%11==0) return true;

        int revNum = 0, oriNum = number;
        while (number > 0) {
            revNum = revNum * 10 + number % 10;
            number = number / 10;
        }
        return revNum == oriNum;
    }
}

File: 1/solution.kt
class Solution {
    fun isPalindrome(number: Int): Boolean {
        val numberToCharList = number.toString().toList()
        val numberToListSize = numberToCharList.size

        return if (numberToListSize % 2 == 0) {
            val splitRight = numberToCharList.subList(0, numberToListSize / 2)
            val splitLeft = numberToCharList.subList(numberToListSize / 2, numberToListSize)

            splitRight == splitLeft.reversed()
        } else {
            val splitRight = numberToCharList.subList(0, numberToListSize / 2)
            val splitLeft = numberToCharList.subList(numberToListSize / 2 + 1, numberToListSize)

            splitRight == splitLeft.reversed() && numberToListSize != 1
        }
    }
}
